apps/web/src/pages/EmployeeProfile/components/GoalsWorkStyleSection/GoalsWorkStyleSection.tsx

```diff
@@ -123,12 +122,8 @@ const GoalsWorkStyleSection = ({
     learningGoals,
     workStyle,
   }: FormValues) => {
-    if (!userAuthInfo?.id) {
-      throw new UnauthorizedError();
-    }
-
     return executeMutation({
-      id: userAuthInfo?.id,
+      id: userId,
```

Comment:
Is there a reason we switched from just reading the user ID from the already queried auth data to just drilling it down?

Reply:
Short answer is typescript wants a string not a string | undefined. That's why someone added the check earlier.

```ts   
if (!userAuthInfo?.id) {
   throw new UnauthorizedError();
}
```

I can't remember if there's another way we were doing this?

Reply:
You can just throw an error to prevent it and create a type guard

Reply:
Thanks for switching back to the useAuth hook and using the `if` to narrow the type.